backend/settings.py

```diff
@@ -210,3 +214,27 @@
         "toolbar": "formatselect | bold italic backcolor | "
             "bullist numlist outdent indent | removeformat | help",
     }
+
+# OAuth Settings, get these from the environment
+# TODO: Add some of these to the Django LTI model
+CANVAS_OAUTH_CLIENT_ID = os.getenv('CANVAS_OAUTH_CLIENT_ID', 'canvas_app_explorer')
+CANVAS_OAUTH_CLIENT_SECRET = os.getenv('CANVAS_OAUTH_CLIENT_SECRET', 'canvas_app_explorer')
+CANVAS_OAUTH_CANVAS_DOMAIN = os.getenv('CANVAS_OAUTH_CANVAS_DOMAIN', 'canvas.instructure.com')
+CANVAS_OAUTH_SCOPES = os.getenv('CANVAS_OAUTH_SCOPES', '').split(',')
+CANVAS_OAUTH_TOKEN_EXPIRATION_BUFFER = os.getenv('CANVAS_OAUTH_TOKEN_EXPIRATION_BUFFER', timedelta())
+CANVAS_OAUTH_ERROR_TEMPLATE = os.getenv('CANVAS_OAUTH_ERROR_TEMPLATE', 'canvas_app_explorer/oauth_error.html')
+
+# These are mostly needed by Canvas but it should also be in on general
+CSRF_COOKIE_SECURE = os.getenv("CSRF_COOKIE_SECURE", False)
+if CSRF_COOKIE_SECURE:
+    CSRF_TRUSTED_ORIGINS = os.getenv("CSRF_TRUSTED_ORIGINS", [])
```

Comment:
Would we need to split this? Any reason for providing this escape hatch anyway?

Reply:
No, right I don't think it matters. I can change this.

Reply:
I'm going to look at this further in another issue when we work on the Canvas API and add/remove tool.